src/components/calcite-inline-editable/calcite-inline-editable.tsx

```diff
@@ -0,0 +1,265 @@
+              <calcite-button
+                appearance="transparent"
+                aria-label={this.intlEnableEditing}
+                class="calcite-inline-editable-enable-editing-button"
+                color="dark"
+                disabled={this.inputElement.disabled}
+                iconStart="pencil"
+                onClick={this.enableEditingHandler}
+                ref={(el) => (this.enableEditingButton = el)}
+                scale={this.scale}
+                theme={this.theme}
+              />
+            )}
+            {this.shouldShowControls && [
+              <div class="calcite-inline-editable-cancel-editing-button-wrapper">
+                <calcite-button
+                  appearance="transparent"
+                  aria-label={this.intlCancelEditing}
+                  class="calcite-inline-editable-cancel-editing-button"
+                  color="dark"
+                  disabled={this.inputElement.disabled}
+                  iconStart="x"
+                  onClick={this.cancelEditingHandler}
+                  scale={this.scale}
+                  theme={this.theme}
+                />
+              </div>,
+              <calcite-button
+                appearance="solid"
+                aria-label={this.intlConfirmChanges}
+                class="calcite-inline-editable-confirm-changes-button"
+                color="blue"
+                disabled={this.inputElement.disabled}
+                iconStart="check"
+                loading={this.loading}
+                onClick={this.confirmChangesChangesHandler}
+                scale={this.scale}
+                theme={this.theme}
+              />
+            ]}
+          </div>
+        </div>
+      </Host>
+    );
+  }
+
+  //--------------------------------------------------------------------------
+  //
+  //  Events
+  //
+  //--------------------------------------------------------------------------
+
+  @Event() calciteInlineEditableCancelEditing: EventEmitter;
+
+  @Event() calciteInlineEditableConfirmChanges: EventEmitter;
+
+  @Event() calciteInlineEditableEnableEditing: EventEmitter;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Event Listeners
+  //
+  //--------------------------------------------------------------------------
+
+  @Listen("calciteInputBlur")
+  blurHandler(): void {
+    if (!this.hasControls) this.disableEditing();
+  }
+
+  @Listen("click", { target: "window" })
+  handleLabelFocus(e: CustomEvent): void {
+    const htmlTarget = e.target as HTMLElement;
+    if (
+      !(
+        htmlTarget.parentElement.tagName === "LABEL" ||
+        htmlTarget.parentElement.tagName === "CALCITE-LABEL"
+      )
+    )
+      return;
+    if (!htmlTarget.parentElement.contains(this.el)) return;
+    e.preventDefault();
+    e.stopPropagation();
+    if (this.editingEnabled) {
+      this.inputElement.setFocus();
+    } else {
+      this.enableEditingButton.setFocus();
+    }
```

Comment:
I'm assuming clicks on the outer label automatically focus the enabled input, so could this logic be simplified to something like?
```
if (!this.editingEnabled) {
  this.enableEditingButton.setFocus();
}
```
If I'm wrong about the automatic focusing, you can ignore this, just thought I would ask.

Reply:
Auto focus was ending up on the confirm/save button when editing was enabled, hence the conditional.